main.go

```diff
@@ -94,25 +94,44 @@ func deduplicate(data []string) []string {
 	return result
 }
 
-func advertise(ingresses []networkingv1.Ingress) error {
+func generateHosts(ingresses []networkingv1.Ingress) ([]string, string, error) {
```

Comment:
Not sure on the `[]string, string, error` return honestly, but for speed I did it like that :smile: 

Happy to change it into some encompassing `struct` though.

Also a little unsure on the naming, `generateHosts` doesn't seem _quite_ right, but felt close enough.

Reply:
no worries; it's not like this project had much forethought anyway; I clobbered it up together quickly

Reply:
go has support for named return values:

```go
func generateHosts(ingresses []networkingv1.Ingress) (names []string, ip string, err error) {
```

this feature has a bad rap because it also allows bare `return` which is often considered harmful. But IMO using named return values as a way to document the meaning of the return values is not a bad idea

Reply:
Agreed, I think that makes sense here. 

I'll add that in before I convert to reviewable 💯